silverback/persistence.py

```diff
@@ -0,0 +1,263 @@
+import pickle
+from abc import ABC, abstractmethod
+from datetime import datetime, timezone
+from typing import Annotated, Optional
+
+from ape.logging import logger
+from pydantic import BaseModel
+from taskiq import TaskiqResult
+from typing_extensions import Self  # Introduced 3.11
+
+from .types import SilverbackIdent
+
+
+class SilverbackState(BaseModel):
+    instance: str
+    network: str
+    # Last block number seen by runner
+    last_block_seen: int
+    # Last block number processed by a worker
+    last_block_processed: int
+    updated: datetime
+
+
+class HandlerResult(BaseModel):
```

Comment:
`taskiq.TaskiqResult` is a Generic `BaseModel` type, where the return value from the inner function is what is generic

If the generic return type can be resolved (potentially from looking at type hints for the return type of a function maybe?) we can probably directly serialize the result type into a database

Reply:
There's extra metadata that is necessary for this persistence layer that isn't part of the TaskiqResult (e.g. instance identifiers and block number).  Seems like just an extra layer of object nesting to not unpack the taskiq object.

As for handler return types, that's another ball of wax that was causing me issues.  For instance, mongo doesn't handle big ints.  Since handlers can return basically anything, I thought pickled data was the best call.  

It did get a bit annoying to go from `TaskiqResult` -> `HandlerResult` -> `HandlerResultDoc`, so def open to other serialization ideas.

Reply:
I did end up subclassing `TaskiqResult`.  It's cleaner.  Though if it changes shape at all, it'll be a wrench in any storage implementations.  Should maybe consider keeping taskiq pinned.

As for the return value, I'm just using Python's `json` module to serialize it in `SQLitePersistentStorage`.